### PR TITLE
feat(python): implement word-ladder with BFS solution

### LINKED ISSUE #58: [Algo] Word Ladder
## Problem: Word Ladder

Find the shortest transformation sequence from beginWord to endWord.

Given two words, beginWord and endWord, and a dictionary wordList, return the number of words in the shortest transformation sequence from beginWord to endWord, or 0 if no such sequence exists.

In each step of the transformation, you must change exactly one letter, and each transformed word must exist in the word list.

### Constraints
- 1 <= beginWord.length <= 10
- endWord.length == beginWord.length
- 1 <= wordList.length <= 5000
- wordList[i].length == beginWord.length
- beginWord, endWord, and wordList[i] consist of lowercase English letters
- beginWord \!= endWord
- All the words in wordList are unique

### Example
Input: beginWord = "hit", endWord = "cog", wordList = ["hot","dot","dog","lot","log","cog"]
Output: 5 (hit -> hot -> dot -> dog -> cog)

### Complexity
- Time: O(N * L * 26) where N is number of words, L is length
- Space: O(N * L)

## Contribution Guidelines
- Implement in one language (we'll add others later)
- Include README with problem statement and test cases
- Add test harness in your implementation
- Follow code style guide in .gemini/styleguide.md

### PR DESCRIPTION
**Pull request by TechGenie-awake**
# Pull Request

## Summary
Implements Word Ladder problem using BFS algorithm to find the shortest transformation sequence from beginWord to endWord.

Closes #58

## Checklist
- [x] Correct folder: `<difficulty>/<problem-slug>/<language>/`
- [x] 3–5 test cases included in problem README or tests
- [x] Time/space complexity documented
- [x] Lint/format pass for this language
- [x] Commit message follows Conventional Commits
- [x] No extraneous files

## Testing
- [x] Verified with sample cases
- [x] Edge cases considered
- [x] If applicable, added tests

## Additional notes
- Includes comprehensive test suite with pytest
- All edge cases covered: missing end word, no valid paths, single characters
- Clean, well-commented code following Python style guide
- README includes problem statement, examples, constraints, and complexity analysis

### CODE REVIEW

File: hard/word-ladder/python/solution.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    queue = deque([(begin_word, 1)])
+    visited = {begin_word}
+
+    while queue:
+        current_word, level = queue.popleft()
+
+        # Try changing each character position
+        for i in range(len(current_word)):
+            # Try all 26 lowercase letters
+            for c in "abcdefghijklmnopqrstuvwxyz":
```
[the remark this answers is not in the document]
Comment (TechGenie-awake) [reply hard/word-ladder/python/solution.py]: Great catch! Updated to use `string.ascii_lowercase` and added the import. Much cleaner and more Pythonic. Thanks!

**Review by TechGenie-awake — COMMENTED** [review]
Addressed ✓

File: hard/word-ladder/python/solution.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        print(f"{status} Test {i}: {begin} -> {end}")
+        print(
+            f"  Original: {original}, Bidirectional: {bidirectional}, Expected: {expected}"
+        )
+
+    if all_passed:
+        print("\n✅ All tests passed for both implementations!")
+    else:
+        print("\n❌ Some tests failed!")
```
[the remark this answers is not in the document]
Comment (TechGenie-awake) [reply hard/word-ladder/python/solution.py]: ✅ Fixed - Removed `word_ladder_original` function and `if __name__ == "__main__"` test harness. All tests are now in the dedicated pytest suite.

**Comment by TechGenie-awake** [comment]
@gemini-code-assist All feedback addressed! ✅

**Changes made:**
- ✅ README updated to describe bidirectional BFS algorithm
- ✅ Type hints made more specific for better clarity
- ✅ Removed redundant test code from solution.py

All tests passing in pytest suite. Ready for review!

**Review by TechGenie-awake — COMMENTED** [review]
Addressed

**Review by 0xReLogic — APPROVED** [review]
Excellent implementation using bidirectional BFS. All 7 test cases pass. Code is well-documented with proper type hints. Black and ruff checks pass. Closes #58.
